server/tests/endpoints/certification/test_certification.py

```diff
@@ -429,3 +429,93 @@ def test_all_criteria_matched(generator: DataGenerator, test_client: TestClient)
             }
         ],
     }
+
+
+def test_bios_is_none(generator: DataGenerator, test_client: TestClient):
+    """
+    Test that Bios information is not required to be sent (some devices may not have it)
+    """
+    vendor = generator.gen_vendor(name="Known Vendor")
+    release = generator.gen_release()
+    machine = generator.gen_machine(
+        canonical_id="202401-00001",
+        configuration=generator.gen_configuration(
+            name="config",
+            platform=generator.gen_platform(name="platform", vendor=vendor),
+        ),
+    )
+    certificate = generator.gen_certificate(machine, release)
+    report = generator.gen_report(
+        certificate,
+        generator.gen_kernel(),
+    )
+    processor = generator.gen_device(
+        vendor=generator.gen_vendor(name="Intel Corp."),
+        name="Intel(R) Core(TM) i5-7300U CPU @ 2.60GHz",
+        bus=BusType.dmi,
+        category=DeviceCategory.PROCESSOR,
+        identifier="dmi:1111",
+        codename="Kaby Lake",
+        reports=[report],
+    )
+    board = generator.gen_device(
+        vendor,
+        identifier="dmi:0001",
+        name="Matching Board",
+        version="v1.0",
+        category=DeviceCategory.BOARD,
+        bus=BusType.dmi,
+        reports=[report],
+    )
+
+    response = test_client.post(
+        "/v1/certification/status",
+        json={
+            "vendor": vendor.name,
+            "model": "Model with Disqualifying Hardware",
+            "architecture": "amd64",
+            "board": {
+                "manufacturer": vendor.name,
+                "product_name": board.name,
+                "version": board.version,
+            },
+            "os": {
+                "distributor": "Canonical Ltd.",
+                "version": release.release,
+                "codename": release.codename,
+                "kernel": {"name": None, "version": "5.7.1-generic", "signature": None},
+            },
+            "processor": {
+                "codename": processor.codename,
+                "frequency": 2.0,
+                "manufacturer": "Intel Corp.",
+                "version": "Intel(R) Core(TM) i7-7600U CPU @ 2.40GHz",
```

Comment:
Just a question. I see here that the processor version `ntel(R) Core(TM) i7-7600U CPU @ 2.40GHz` is different from the processor name specified above `Intel(R) Core(TM) i5-7300U CPU @ 2.60GHz`. Yet the resulting status is `Certified`. Does that mean that the processor doesn't need to be exactly the same to be considered certified?

Reply:
we match processor codename, not the exact model name. So, it means that the processor version (model name) can be different, but codename must be the same
you can check it here: https://github.com/canonical/hardware-api/blob/1bb344/server/hwapi/endpoints/certification/logic.py#L55-L59

Reply:
Got it, thanks for the clarification